src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IThrowOperation.cs

```diff
@@ -0,0 +1,1918 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ThrowFlow_01()
+        {
+            var source = @"
+class C
+{
+    void F()
+    /*<bind>*/{
+        throw;
+    }/*</bind>*/
+}";
+
+            var compilation = CreateCompilation(source, parseOptions: TestOptions.RegularWithFlowAnalysisFeature);
+
+            compilation.VerifyDiagnostics(
+                // (6,9): error CS0156: A throw statement with no arguments is not allowed outside of a catch clause
+                //         throw;
+                Diagnostic(ErrorCode.ERR_BadEmptyThrow, "throw").WithLocation(6, 9)
+                );
+
+            string expectedGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (0)
+    Next (ReThrow) Block[null]
+Block[B2] - Exit
```

Comment:
Is this block needed?

Reply:
> Is this block needed?

Yes, each graph should start with an entry block and end with an exit block. Even when the exit block is unreachable.

---
In reply to: [173523696](https://github.com/dotnet/roslyn/pull/25338#discussion_r173523696) [](ancestors = 173523696)

Reply:
+1 on retaining these blocks - they are very beneficial in dataflow analysis where pre and post analysis for data associated with parameters (especially ref and out parameters) and return value is done during analysis of entry/exit blocks.